File: python/line-up/line_up.py
"""
Greet customers in a line up
"""

def line_up(name, number):
    """
    Compute the desired greeting for a customer with a given name and number
    name: The name of the customer
    number: The ordering of the customer for the given day
    returns: Formatted string greeting the customer with a number and its proper suffix
    """
    tens = number % 100
    ones = number % 10
    suffix = 'th'
    if ones == 1 and tens != 11:
        suffix = 'st'
    if ones == 2 and tens != 12:
        suffix = 'nd'
    if ones == 3 and tens != 13:
        suffix = 'rd'
    
    return f'{name}, you are the {number}{suffix} customer we serve today. Thank you!'

File: php/line-up/LineUp.php
<?php
// Line-up exercise address a customer by name and number with number having a nd, rd, th, or st suffix

declare(strict_types=1);

// Address a customer by their name and number where the number has the proper suffix.
// @param $name: The name of the customer
// @param $number: The customer's position in line (integer >= 1)
// @returns: A custom greeting for the customer.
function format(string $name, int $number): string
{
    $last_digit = $number % 10;
    $last_two_digits = $number % 100;

    $suffix = match ($number % 100) {
        11, 12, 13 => "th", 
        default => match($number % 10) {
            1 => "st", 
            2 => "nd", 
            3 => "rd", 
            default => "th",
        }
    };
    return "{$name}, you are the {$number}{$suffix} customer we serve today. Thank you!";
}
